scheduler/queue/queue.go

```diff
@@ -209,8 +212,22 @@ func (q *queue) signal(ctx context.Context) error {
 			// 		Msg("cannot update queue item")
 			// 	continue
 			// }
-			select {
-			case w.channel <- item:
+
+			// TODO: refactor to its own unexported method
+			sendWork := func() bool {
+				expiration := time.NewTimer(1 * time.Minute)
+				defer expiration.Stop()
+				select {
+				case w.channel <- item:
+					return true
+				case <-w.done:
+					// Worker will exit when we call the deferred q.Unlock()
+				case <-expiration.C:
+					// Worker failed to ack before timeout
+				}
```

Comment:
The `expiration` variable isn't necessary and the duration should be `q.interval`.
```go
sendWork := func() bool {
	select {
	case w.channel <- item:
		return true
	case <-w.done:
		// Worker will exit when we call the deferred q.Unlock()
	case <-time.After(q.interval):
		// Worker failed to ack before timeout
	}
	return false
}
```